libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -275,7 +275,7 @@ bool ExpressionCompiler::visit(Assignment const& _assignment)
 		solAssert(*_assignment.annotation().type == leftType, "");
 	bool cleanupNeeded = false;
 	if (op != Token::Assign)
-		cleanupNeeded = cleanupNeededForOp(leftType.category(), binOp);
+		cleanupNeeded = m_context.arithmetic() == Arithmetic::Checked || cleanupNeededForOp(leftType.category(), binOp);
```

Comment:
Is this just extra safety?

Reply:
Same as https://github.com/ethereum/solidity/pull/9465#discussion_r505664925 - since the arithmetic functions in yul already do the cleanup, we should probably not clean up before them.